cmd/runner.go

```diff
@@ -0,0 +1,44 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/cobra"
+)
+
+type runner struct {
+	flag   *flag
+	logger micrologger.Logger
+	stdout io.Writer
+	stderr io.Writer
+}
+
+func (r *runner) Run(cmd *cobra.Command, args []string) {
```

Comment:
I think we can have some `RunE` or `RunError` interface to do the logging and process termination in fewer places.

Reply:
That makes sense but where should it live? microkit?

Reply:
So when this is the method which is called by cobra then it is done there. Good question what happens on error then, but there is something like `RunE` and I thought we should rather use this and do normal error handling here.

Reply:
Got it. Yes using `RunE` is nice and the error is passed to Execute when it's called. I renamed `Run` to `RunWithError` to align with mainWithError.

@xh3b4sd PTAL